Lib/test/test_c_locale_coercion.py

```diff
@@ -296,26 +307,26 @@ def _check_c_locale_coercion(self,
             "LC_ALL": "",
         }
         base_var_dict.update(extra_vars)
-        for env_var in ("LANG", "LC_CTYPE"):
-            for locale_to_set in ("", "C", "POSIX", "invalid.ascii"):
-                # XXX (ncoghlan): *BSD platforms don't behave as expected in the
-                #                 POSIX locale, so we skip that for now
-                # See https://bugs.python.org/issue30672 for discussion
-                if locale_to_set == "POSIX":
-                    continue
-
-                # Platforms using UTF-8 in the C locale do not print
-                # CLI_COERCION_WARNING when all the locale envt variables are
-                # not set or set to the empty string.
-                _expected_warnings = expected_warnings
-                for _env_var in base_var_dict:
-                    if base_var_dict[_env_var]:
-                        break
-                else:
-                    if (C_LOCALE_STREAM_ENCODING == "utf-8" and
-                           locale_to_set == "" and coerce_c_locale == "warn"):
-                        _expected_warnings = None
 
+        # Check behaviour for the default locale
+        with self.subTest(default_locale=True,
```

Comment:
This block can only be run if the following conditional is true, otherwise some test results may not match the expected results:
```
if set(base_var_dict).isdisjoint(extra_vars):
```
For clarity, this statement written above may be moved down, when checking for explicitly configured locales:
```
base_var_dict.update(extra_vars)
```

Reply:
The expectation is that the test case that passes in `extra_vars` will also set `expected_warnings` and `coercion_expected` appropriately.

The current use case is the tests that sets `LC_ALL='C'`, and expects that to both disable coercion, and potentially result in a warning (for the `PYTHONCOERCECLOCALE=warn` case).